core/src/avm1/globals/sound.rs

```diff
@@ -378,10 +378,13 @@ fn start<'gc>(
         .unwrap_or(&Value::Number(1.0))
         .coerce_to_f64(activation)?;
 
-    let loops = if loops >= 1.0 && loops <= f64::from(std::i16::MAX) {
-        loops as u16
-    } else {
+    // TODO: Handle loops > std::u16::MAX.
+    let loops = if loops < 1.0 {
         1
+    } else if loops > f64::from(std::u16::MAX) {
+        std::u16::MAX
+    } else {
+        loops as u16
```

Comment:
Float-to-int cast to Rust is now defined to saturate, i.e. `999999.0 as u16` will automatically be `std::u16::MAX`. This wasn't the case years ago when this code was written. So don't have to explicitly check for it, can remove `loops <= f64::from(std::i16::MAX)` from the original condition.